### DISCUSSION TITLE
Google Analytics

### DISCUSSION THREAD

**Discussion by shinanqu** [oleeskild/obsidian-digital-garden#195] [Q&A] [answered]
Hello,

I'm wondering if it's possible to implement analytics/stats for the digital garden. Doesn't need to be Google, any other (simple) solution would be appreciated as well.

Thank you.

**Comment by oleeskild** [comment, marked as answer]
If you are using Vercel, you can add a custom snippet using [slots](https://dg-docs.ole.dev/advanced/adding-custom-components/), in the following path on your site `src/site/_includes/components/user/common/footer/analytics.njk`
```html
<script defer src="/_vercel/insights/script.js"></script>
```

Data should then show up here: 
![CleanShot 2023-02-15 at 09 04 24@2x](https://user-images.githubusercontent.com/6201338/218968446-8b582f70-e12c-4f34-8653-7b852967bb7a.png)

Let me know if you need any help with it, or if you get it to work.

**Reply by gtewary1** [reply]
Could you please elaborate as to how exactly one adds the script?
Is it done in Obsidian or GitHub?

**Reply by shinanqu** [reply]
I also had some issues. I managed to implement Google Analytics, but the Vercel analytics are not running.

_- Could you please elaborate as to how exactly one adds the script?_

I added it to Github at the described path. The Vercal analytics changed to "read to start collecting Analytics", but no data is collected (I tried with different browser, without ad-block, etc.) The Google Script is working, though.

![2023_03_09_1207-V](https://user-images.githubusercontent.com/46276550/223926215-43d39893-06cb-4851-aff7-89d7acd0cf01.jpg)

![2023_03_09_1210-k](https://user-images.githubusercontent.com/46276550/223926362-ca4dc572-e19a-40cd-bcaf-a482f1c3fda5.jpg)


![2023_03_09_1207-D](https://user-images.githubusercontent.com/46276550/223926205-9d9025f4-64f9-4eac-ad8c-031d868a56f1.jpg)

**Reply by oleeskild** [reply]
@shinanqu Are you looking at the "Web Vitals" tab or the "Audiences" tab under Analytics? It's a bit confusing, but Web Vitals is something else but is the default view when visiting the Vercel Analytics tab. Look at the second arrow in the image in the original answer to see where to click.

**Reply by shinanqu** [reply]
Oh yes, my mistake, I was in the wrong tab. But anyway, I don't receive any data. Google PageSpeed Insights also displays the following error message:
![2023_03_09_2322-6](https://user-images.githubusercontent.com/46276550/224087260-c227aed4-a30d-4309-91b1-41db7422a2db.jpg)

**Reply by oleeskild** [reply]
Hmm. Have you clicked this? 
![CleanShot 2023-03-09 at 19 12 17@2x](https://user-images.githubusercontent.com/6201338/224117851-a7800177-c43f-4f9e-adf1-45c031ef35fe.png)

**Reply by shinanqu** [reply]
Yes, it is clicked. But well, it's not an urgent issue (at least not for me) because Google Analytics is working just fine.

![2023_03_10_1037-V](https://user-images.githubusercontent.com/46276550/224217641-ea90430a-63c0-45a8-b46e-8a45c2232f1c.jpg)

![2023_03_10_1038-E](https://user-images.githubusercontent.com/46276550/224217651-8fa74134-0a71-4c95-88b4-2c04982a8bea.jpg)

**Comment by shinanqu** [comment]
Great, thank you so much for your quick answer!

**Reply by shinanqu** [reply]
It is working!!! I have set it up around 5-6 days ago and no matter how often I visited my homepage (VPN, different browsers, etc.) there was no visitor registered. But finally, 5 days later, Vercel received the first data. Not sure why it took so long, but now everything is working.

If anybody else has set it up correctly and it isn't working, perhaps you also have to be a bit patient.

Thank you Ole for your help!

**Reply by dominoto** [reply]
Hey, just wanted to chime in. I was just playing with this and uBlock Origin was blocking the script. After disabling uBlock for my site, analytics started getting the data.
Also tried to enable Speed Insights in the same way as @oleeskild described and it works! Just add `<script defer src="/_vercel/speed-insights/script.js"></script>` in the same `analytics.njk` file.

**Reply by ZGChung** [reply]
Hello there. I also wanted to enable Vercel Analytics. After some searching, I found this discussion, but I could not follow the solution proposed above. Now inside the /src/site/_includes/components/ folder, there is no more folder called "user". Also, I couldn't find any file called "analytics.njk" now. Maybe the way of enabling Vercel Analytics has been modified. 

On Vercel I saw the instruction like this:
<img width="1214" alt="image" src="https://github.com/oleeskild/obsidian-digital-garden/assets/69182806/48ff6714-7622-40c3-aba9-e97dd019ca16">

However, I do not really know where to input the command "npm i @vercel/analytics". Because I used the one-click deploy with the github template of Digital Garden, thus I don't know where to open a terminal, sorry.

I really appreciate your help, thanks in advance!

**Reply by dominoto** [reply]
@ZGChung you need to create the missing folders and empty `analytics.njk` file yourself so the full path becomes like this: `src/site/_includes/components/user/common/footer/analytics.njk`. After that put these lines into `analytics.njk`:
```
<script defer src="/_vercel/insights/script.js"></script>
<script defer src="/_vercel/speed-insights/script.js"></script>
```
That is what worked for me. You can do it all directly in your GitHub project.

**Reply by ZGChung** [reply]
Worked well. Thanks very much for the help!
